src/Uno.UI/UI/Xaml/UIElement.Pointers.cs

```diff
@@ -436,6 +459,163 @@ private protected void CompleteGesture()
+				|| !args.Pointer.Properties.IsLeftButtonPressed)
+			{
+				//if (CoreWindow.GetForCurrentThread()!.LastPointerEvent is PointerRoutedEventArgs ptArgs
+				//	&& ptArgs.Pointer.PointerDeviceType == args.PointerDeviceType)
+				//{
+				//	if (args.DraggingState == )
+				//	Window.Current._dragDropManager.ProcessPointerMoved();
+				//}
+
+				return;
+			}
+
+			// Note: We do not provide the _pendingRaisedEvent.args since it has probably not been updated yet,
+			//		 but as we are in the handler of an event from the gesture recognizer,
+			//		 the LastPointerEvent from the CoreWindow will be up to date.
+			StartDragAsyncCore(args.Pointer);
+		}
+
+		public IAsyncOperation<DataPackageOperation> StartDragAsync(PointerPoint pointerPoint)
+			=> StartDragAsyncCore(pointerPoint, _pendingRaisedEvent.args).AsAsyncOperation();
+
+		private Task<DataPackageOperation> StartDragAsyncCore(PointerPoint pointer, PointerRoutedEventArgs args = null)
+		{
+			args ??= CoreWindow.GetForCurrentThread()!.LastPointerEvent as PointerRoutedEventArgs;
+			if (args is null || args.Pointer.PointerDeviceType != pointer.PointerDevice.PointerDeviceType)
+			{
+				// Fairly impossible case ...
+				return Task.FromResult(DataPackageOperation.None);
+			}
+
+			// Note: originalSource = this => DragStarting is not actually a routed event, the original source is always the sender
+			var routedArgs = new DragStartingEventArgs(this, args);
+			PrepareShare(routedArgs.Data); // Gives opportunity to the control to fulfill the data
+			SafeRaiseEvent(DragStartingEvent, routedArgs); // The event won't bubble, cf. PrepareManagedDragAndDropEventBubbling
+
+			// TODO: Add support for the starting deferral!
+
+			if (routedArgs.Cancel)
+			{
+				// The completed event is not raised if the starting has been cancelled
+				return Task.FromCanceled<DataPackageOperation>(CancellationToken.None);
+			}
+
+			if (!pointer.Properties.HasPressedButton)
+			{
+				// This is the UWP behavior: if no button is pressed, then the drag is completed immediately
+				var noneResult = Task.FromResult(DataPackageOperation.None);
+				OnDragCompleted(noneResult, this);
+				return noneResult;
+			}
+
+			var result = new TaskCompletionSource<DataPackageOperation>();
+			result.Task.ContinueWith(OnDragCompleted, this, TaskContinuationOptions.NotOnCanceled | TaskContinuationOptions.RunContinuationsAsynchronously);
+
+			var dragInfo = new CoreDragInfo(
+				routedArgs.Data.GetView(),
+				routedArgs.AllowedOperations,
+				dragUI: routedArgs.DragUI,
+				pointer: pointer.PointerDevice);
+			dragInfo.RegisterCompletedCallback(result.SetResult);
+
+			CoreDragDropManager.GetForCurrentView()!.DragStarted(dragInfo);
+
+			return result.Task;
+		}
+
+		private static void OnDragCompleted(Task<DataPackageOperation> resultTask, object snd)
+		{
+			var that = (UIElement)snd;
+			var result = resultTask.IsFaulted
+				? DataPackageOperation.None
+				: resultTask.Result;
```

Comment:
Codacy found an issue: [Replace this use of 'Task.Result' with 'await'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6270104)